scripts/check-mcp-compliance.py

```
@@ -0,0 +1,308 @@
+        content = data_file.read_text()
+
+        # Should NOT return simulated results
+        if "simulated results" in content.lower() and "NO SIMULATIONS" not in content:
+            issues.append("Data processor still returns simulated results")
+    else:
+        issues.append("Data processor file not found")
+
+    return len(issues) == 0, issues
+
+
+def run_all_checks() -> bool:
```

Ruff (0.11.9)

`run_all_checks` is too complex (15 > 10)

(C901)